src/App.tsx

```diff
@@ -1,14 +1,69 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
+import { getTodos } from './api';
+
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { TodoModal } from './components/TodoModal';
+
+import { Todo } from './types/Todo';
+import { Status } from './types/Status';
+
+function getFilteredTodos(
+  todos: Todo[],
+  query: string,
+  status: Status,
+) {
+  let visibleTodos = [...todos];
+
+  if (query) {
+    visibleTodos = visibleTodos.filter(
+      todo => todo.title.toLocaleLowerCase()
+        .includes(query.trim().toLocaleLowerCase()),
+    );
+  }
+
+  return visibleTodos
+    .filter(todo => {
+      switch (status) {
+        case Status.COMPLETED:
+          return todo.completed;
+
+        case Status.ACTIVE:
+          return !todo.completed;
+
+        default:
+          return true;
+      }
+    });
```

Comment:
move filter inside switch for avoiding redindant iterration in default case.
```suggestion
      switch (status) {
        case Status.COMPLETED:
          return visibleTodos.filter(todo => todo.completed);

        case Status.ACTIVE:
          return visibleTodos.filter(todo => !todo.completed);

        default:
          return visibleTodos;
      }
```